utilities/test/test_trailing_whitespace.py

```diff
@@ -0,0 +1,92 @@
+from os import walk, sep, pardir
+from os.path import split, join, abspath, exists, isfile , basename
+from glob import glob
+import re
+import random
+import ast
+import inspect
+ 
+_failed_expectations = []
+ 
+def _log_failure(msg=None):
+    (filename, line, funcname, contextlist) =  inspect.stack()[2][1:5]
+    filename = basename(filename)
+    context = contextlist[0]
+    _failed_expectations.append('%s\n' % 
+        ((('%s' % msg) if msg else '')))
+ 
+def _report_failures():
+    global _failed_expectations
+    report = []
+    if _failed_expectations:
+        (filename, line, funcname) =  inspect.stack()[1][1:4]
+        report = [
+            'Failed Expectations:%s\n' % len(_failed_expectations)]
```

Comment:
The code is expected to have no trailing whitespace. Each trailing whitespace is a failed expectation